src/back/glsl/mod.rs

```diff
@@ -126,14 +123,16 @@ impl fmt::Display for Version {
 
 /// Structure that contains the configuration used in the [`Writer`](Writer)
 #[derive(Debug, Clone)]
-pub struct Options {
+pub struct Options<'a> {
     /// The glsl version to be used
     pub version: Version,
-    /// The name and stage of the entry point
+    /// The stage of the entry point
+    pub shader_stage: ShaderStage,
+    /// The name of the entry point
     ///
     /// If no entry point that matches is found a error will be thrown while creating a new instance
     /// of [`Writer`](struct.Writer.html)
-    pub entry_point: (ShaderStage, String),
+    pub entry_point: &'a str,
```

Comment:
I'm not sure this is needed to borrow. This makes it impossible to deserialize `Options`, and we'd ideally want to use it directly in those `.param.ron` files.

Reply:
It's possible to deserialize a struct with borrows

Reply:
My concern here, and it's a minor-ish concern, is that you have to carry that `'options` lifetime through the code. It's polluting the definitions, and it doesn't get us anything. So the old choice with `String` was both simpler and more in-line with other backends.